### PR TITLE
Srsuddath search UI overhaul 24988

### PR DESCRIPTION
**Pull request by srsuddath**
## Description
This PR accomplishes a number of tickets, listed below:
https://github.com/department-of-veterans-affairs/va.gov-team/issues/19225
https://github.com/department-of-veterans-affairs/va.gov-team/issues/24988
https://github.com/department-of-veterans-affairs/va.gov-team/issues/24118
https://github.com/department-of-veterans-affairs/va.gov-team/issues/24968
https://github.com/department-of-veterans-affairs/va.gov-team/issues/24778
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26287
https://github.com/department-of-veterans-affairs/va.gov-team/issues/26506

## Testing done
manual and review instance testing (still in progress)

## Acceptance criteria
- [x] UI passes consistency checks
- [x] Spelling correction message is displayed
- [x] Updated error logging is included

## Definition of done
- [X] Events are logged appropriately
- [X] Documentation has been updated, if applicable
- [X] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [X] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/search/styles.scss [hunk 1]
```diff
@@ -128,18 +145,54 @@
...
+    flex-direction: column;
+  }
+
     input,
     button {
       margin: 0;
       max-width: none;
+      font-family: "Source Sans Pro";
+
+      @media (max-width: $small-screen) {
```
Comment (zacharymorel) [inline src/applications/search/styles.scss:158]: Just a small thought here. I was under the _impression_ that we write styles mobile first. I think that is standard for the VA, but I am not I'm not sure where I herd that or saw it. 
Maybe @kelsonic has more input to whether or not there is a suggested common practice.

File: src/applications/search/containers/SearchApp.jsx [hunk 2]
```diff
@@ -199,42 +199,47 @@ class SearchApp extends React.Component {
...
-            data-e2e-id="alert-box"
-          />
+          {/* this is the alert box for when searches fail due to server issues */}
+          <va-alert status="error" data-e2e-id="alert-box">
+            <h3 slot="headline">Your search didn't go through</h3>
+            <div>
+              We’re sorry. Something went wrong on our end, and your search
+              didn't go through. Please try again
+            </div>
+          </va-alert>
```
Comment (zacharymorel) [inline src/applications/search/containers/SearchApp.jsx:242]: Oh I have to do this. I might shamelessly peek at this haha.

File: src/applications/search/styles.scss (hunk 1 above)
Comment (srsuddath) [reply src/applications/search/styles.scss:158]: good to know! This is all legacy code so I was tweaking and adding to existing. happy to rewrite some stuff to make it fit our standards!

File: src/applications/search/containers/SearchApp.jsx (hunk 2 above)
Comment (srsuddath) [reply src/applications/search/containers/SearchApp.jsx:242]: fun story, we were using the old alert box set up, and it just wasn't even connected to anything. checking to see if there was an error, but no code was written that would actually populate an error. This seemed as good a time as any to make the switch to actually doing some error handling!

File: src/applications/search/styles.scss (hunk 1 above)
Comment (kelsonic) [reply src/applications/search/styles.scss:158]: @zacharymorel that's correct! 🙂 @media queries are usually for min-width: $medium-screen here I've noticed 👍

**Review by kelsonic — APPROVED** [review]

**Review by k80bowman — APPROVED** [review]

**Pull request merged**